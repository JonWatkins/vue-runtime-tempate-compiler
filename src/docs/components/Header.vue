<template>
  <header
    class="navbar navbar-expand navbar-light flex-column flex-md-row navbar sticky-top"
  >
    <router-link to="/" class="navbar-brand ml-1 mr-0 mr-md-2">
      VueRuntimeTemplateCompiler
    </router-link>
    <ul class="navbar-nav ml-auto">
      <li class="nav-item mt-1">
        <router-link to="/guide/getting-started">
          Guide
        </router-link>
      </li>
      <li class="nav-item  mt-1 ml-3">
        <router-link to="/api-reference">
          API Reference
        </router-link>
      </li>
      <li class="nav-item  mt-1 ml-3">
        <router-link to="/sandbox">
          Sandbox
        </router-link>
      </li>
      <li class="nav-item  mt-1 ml-3">
        <a
          href="https://github.com/JonWatkins/vue-runtime-template-compiler/releases"
          target="_blank"
          class="external"
          rel="nofollow"
        >
          Release Notes<i class="fas fa-external-link-alt" />
        </a>
      </li>
      <li class="nav-item ml-3">
        <a
          href="https://github.com/jonwatkins/vue-runtime-template-compiler"
          target="_blank"
          class="border-bottom-0 icon-hover"
          rel="nofollow"
          alt="GitHub"
        >
          <i class="fab fa-github fa-2x"></i>
        </a>
      </li>
      <li class="nav-item text-muted ml-3">
        <a
          href="https://www.npmjs.com/package/vue-runtime-template-compiler"
          target="_blank"
          class="border-bottom-0 icon-hover"
          rel="nofollow"
          alt="NPM"
        >
          <i class="fab fa-npm fa-2x"></i>
        </a>
      </li>
    </ul>
  </header>
</template>

<script>
import { version } from '../../../package.json'

export default {
  name: 'Header',

  data() {
    return {
      version
    }
  }
}
</script>

<style lang="scss" rel="stylesheet/scss">
.navbar {
  background: white;
  border-bottom: 1px solid #eaecef;

  .navbar-nav {
    a {
      color: #2c3e50;
      border-bottom: 2px solid transparent;
      transition: border 0.25s linear;
      padding-bottom: 3px;

      &.active,
      &:hover {
        border-color: #3eaf7c;
      }
    }
  }
}
</style>
