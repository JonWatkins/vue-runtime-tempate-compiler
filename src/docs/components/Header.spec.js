import { shallowMount } from '@vue/test-utils'
import Header from './Header.vue'
const { localVueInst } = global

let wrapper

describe('Header', () => {
  beforeEach(() => {
    wrapper = shallowMount(Header, {
      localVue: localVueInst
    })
  })

  it('is called', () => {
    expect(wrapper.name()).toBe('Header')
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
