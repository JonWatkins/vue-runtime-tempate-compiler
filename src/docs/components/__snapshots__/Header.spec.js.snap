// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly 1`] = `
<header class="navbar navbar-expand navbar-light flex-column flex-md-row navbar sticky-top">
  <router-link-stub to="/" tag="a" event="click" class="navbar-brand ml-1 mr-0 mr-md-2 ">
    VueRuntimeTemplateCompiler
  </router-link-stub> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div class="collapse navbar-collapse">
    <ul class="navbar-nav ml-auto">
      <li class="nav-item mt-1">
        <router-link-stub to="/guide/getting-started" tag="a" event="click">
          Guide
        </router-link-stub>
      </li>
      <li class="nav-item  mt-1 ml-3">
        <router-link-stub to="/api-reference" tag="a" event="click">
          API Reference
        </router-link-stub>
      </li>
      <li class="nav-item  mt-1 ml-3">
        <router-link-stub to="/sandbox" tag="a" event="click">
          Sandbox
        </router-link-stub>
      </li>
      <li class="nav-item  mt-1 ml-3"><a href="https://github.com/JonWatkins/vue-runtime-template-compiler/releases" target="_blank" rel="nofollow" class="external">
          Release Notes<i class="fas fa-external-link-alt"></i></a></li>
      <li class="nav-item text-muted ml-3 d-none d-sm-block d-sm-none d-md-block"><a href="https://github.com/jonwatkins/vue-runtime-template-compiler" target="_blank" rel="nofollow" alt="GitHub" class="border-bottom-0 icon-hover"><i class="fab fa-github fa-2x"></i></a></li>
      <li class="nav-item text-muted ml-3 d-none d-sm-block d-sm-none d-md-block"><a href="https://www.npmjs.com/package/vue-runtime-template-compiler" target="_blank" rel="nofollow" alt="NPM" class="border-bottom-0 icon-hover"><i class="fab fa-npm fa-2x"></i></a></li>
    </ul>
  </div>
</header>
`;
